lib/matplotlib/axes/_axes.py

```diff
@@ -2581,6 +2582,123 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, labels=None, *, fmt="%g", label_type="edge",
+                  padding=0, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+        You may need to adjust the axis limits to fit the labels.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
```

Comment:
Probably want to reference `bar` and `barh` here:
```suggestion
            Container with all the bars and optionally errorbars, likely returned from `.bar` or `.barh`.
```